### PR TITLE
chore: Clear Up Info Messages for at_widget packages using  flutter 3.23 beta channel

### LINKED ISSUE #859: Clear up `info` messages from `flutter analyze` (Flutter beta 3.23 edition)
### Which at_widget package you find issue with?

at_backupkey_flutter, at_common_flutter, at_contacts_group_flutter, at_events_flutter, at_location_flutter, at_login_flutter, at_notify_flutter, at_onboarding_flutter, at_sync_ui_flutter, at_theme_flutter

### On what platforms did you experience the issue?

Linux

### What version are you using?

Trunk

### What happened?

A new Flutter beta (for 3.23) dropped on 6 Jun (just after we cleared up the infos for Flutter 3.22 in #847)

See recent static analysis runs e.g. https://github.com/atsign-foundation/at_widgets/actions/runs/9446898794

### Steps to reproduce

```bash
1. Go to any of the listed packages
2. Run `flutter analyze` using Flutter 3.23 beta
```

### PR DESCRIPTION
**Pull request by CurtlyCritchlow**
**- What I did**
closes #859 
**- How I did it**
Run flutter analyze to see info message.
Make the changes to clear the info message
**- How to verify it**
Github static analyser should run successfully.
**- Description for the changelog**

Clear up info messages

### CODE REVIEW

**Comment by CurtlyCritchlow** [comment]
@cpswan One of the check failed. This because `onPopInvokedWithResult` is the replacement method on the flutter 3.23 beta  for `onPopInvoked` but `onPopInvokedWithResult` is not available on the stable channel. I'll have to revert to `onPopInvoked` but this will be the only info message on the beta channel.

**Comment by cpswan** [comment]
OK @CurtlyCritchlow

Can we put an annotation in place to supress that info message, and an issue to remind us to make the change once it lands in stable.

[Breaking changes > Generic types in PopScope](https://docs.flutter.dev/release/breaking-changes/popscope-with-result) for reference.

File: packages/at_common_flutter/example/pubspec.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ dependencies:
   # activated in the `analysis_options.yaml` file located at the root of your
   # package. See that file for information about deactivating specific lint
   # rules and activating additional ones.
-  flutter_lints: ^1.0.0
+  flutter_lints: ^2.0.2
```
Comment (cpswan) [inline packages/at_common_flutter/example/pubspec.yaml]: Why not 4.0.0 here?

File: packages/at_events_flutter/example/pubspec.yaml [hunk 2]
```diff
@@ -42,6 +43,8 @@ dependencies:
   cupertino_icons: ^1.0.6
   flutter:
     sdk: flutter
+  flutter_dotenv: ^5.1.0
+  path_provider: ^2.1.3
 
 dev_dependencies:
```
Comment (cpswan) [inline packages/at_events_flutter/example/pubspec.yaml:49]: Very old flutter_lints here too.

File: packages/at_location_flutter/example/pubspec.yaml [hunk 3]
```diff
@@ -43,6 +44,10 @@ dependencies:
   flutter:
     sdk: flutter
   latlong2: ^0.9.0
+  path_provider: ^2.1.3
+  at_common_flutter: ^2.0.13
+  at_lookup: ^3.0.47
+  flutter_dotenv: ^5.1.0
 
 dev_dependencies:
```
Comment (cpswan) [inline packages/at_location_flutter/example/pubspec.yaml:52]: and here

File: packages/at_notify_flutter/example/pubspec.yaml [hunk 4]
```diff
@@ -37,6 +39,7 @@ dependencies:
   cupertino_icons: ^1.0.6
   flutter:
     sdk: flutter
+  path_provider: ^2.1.3
 
 dev_dependencies:
```
Comment (cpswan) [inline packages/at_notify_flutter/example/pubspec.yaml:44]: and here

File: packages/at_sync_ui_flutter/example/pubspec.yaml [hunk 5]
```diff
@@ -29,13 +29,16 @@ environment:
...
+  at_client_mobile: ^3.2.15
   at_onboarding_flutter: ^6.1.7
+  at_sync_ui_flutter: ^1.0.11
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.6
   flutter:
     sdk: flutter
   flutter_colorpicker: ^1.1.0
+  path_provider: ^2.1.3
```
Comment (cpswan) [inline packages/at_sync_ui_flutter/example/pubspec.yaml:42]: and here

File: packages/at_theme_flutter/example/pubspec.yaml [hunk 6]
```diff
@@ -31,13 +31,15 @@ dependencies:
...
   # Use with the CupertinoIcons class for iOS style icons.
+  at_client_mobile: ^3.2.15
   at_common_flutter: ^2.0.12
   at_onboarding_flutter: ^6.1.7
   at_theme_flutter:
     path: ../
   cupertino_icons: ^1.0.6
   flutter:
     sdk: flutter
+  path_provider: ^2.1.3
```
Comment (cpswan) [inline packages/at_theme_flutter/example/pubspec.yaml:43]: and here

**Review by cpswan — DISMISSED** [review]
This may be premature, as the PR is still draft, but I'm happy for this to be merged, so long as your confident with tests.

The one thing I did notice, which can be changed now, or later, or maybe even left, is the example pubspecs still use a very old flutter_lints

**Comment by CurtlyCritchlow** [comment]
@cpswan I'm about to inbox you about a strategy for publishing the new versions of our packages. I didn't update the dependencies as yet because in some cases it cause conflicts due to other at_widgets packages that are dependencies using older versions and not the newer versions.

**Review by cpswan — APPROVED** [review]

**Pull request merged**